back/app/services/authoring_assistance_service.rb

```diff
@@ -0,0 +1,49 @@
+class AuthoringAssistanceService
+  def initialize(authoring_assistance_response)
+    @authoring_assistance_response = authoring_assistance_response
+  end
+
+  def analyze!
+    duplicate_inputs_thread = start_thread { duplicate_inputs_response(_1) }
+    toxicity_thread = start_thread { toxicity_response(_1) }
+    custom_free_prompt_thread = start_thread { custom_free_prompt_response(_1) }
+    [duplicate_inputs_thread, toxicity_thread, custom_free_prompt_thread].each(&:join)
+    prompt_response = {
+      **duplicate_inputs_thread[:response],
+      **toxicity_thread[:response],
+      **custom_free_prompt_thread[:response]
+    }
+    @authoring_assistance_response.update!(prompt_response: prompt_response)
+  end
+
+  private
+
+  def duplicate_inputs_response(authoring_assistance_response)
+    service = SimilarIdeasService.new(authoring_assistance_response.idea)
+    service.upsert_embedding!
+    threshold = 0.4
```

Comment:
We will try to find a way to return only ideas that are "sufficiently" similar next week.